prow/cmd/build/controller.go

```diff
@@ -340,18 +355,35 @@ func prowJobStatus(bs buildv1alpha1.BuildStatus) (prowjobv1.ProwJobState, string
 }
 
 // makeBuild creates a build from the prowjob, using the prowjob's buildspec.
-func makeBuild(pj prowjobv1.ProwJob) (*buildv1alpha1.Build, error) {
+func makeBuild(pj prowjobv1.ProwJob, buildID string) (*buildv1alpha1.Build, error) {
 	if pj.Spec.BuildSpec == nil {
 		return nil, errors.New("nil BuildSpec")
 	}
-	return &buildv1alpha1.Build{
+	rawEnv, err := downwardapi.EnvForSpec(downwardapi.NewJobSpec(pj.Spec, buildID, pj.Name))
+	if err != nil {
+		return nil, fmt.Errorf("failed EnvForSpec: %v", err)
+	}
+	podLabels, annotations := decorate.GetLabelsAndAnnotations(pj)
+	b := buildv1alpha1.Build{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      pj.Name,
-			Namespace: pj.Namespace,
+			Annotations: annotations,
+			Name:        pj.Name,
+			Namespace:   pj.Namespace,
```

Comment:
We should probably use the configured pod namespace here or add configuration for a build-crd namespace. The ProwJob namespace doesn't seem like the correct value. https://github.com/kubernetes/test-infra/blob/1c99d45ba0fa7aa03710fc3b7cbbf1f9714f26be/prow/config/config.go#L85-L89

Reply:
This might be a big enough change to do in a follow up, especially because it would be a breaking change for existing deployments.